javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.results.RunResult;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import scala.collection.Iterator;
+import scala.math.Ordering;
+import scala.math.Ordering$;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.Throughput)
+@Warmup(iterations = 2, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@OutputTimeUnit(TimeUnit.SECONDS)
+@Fork(value = 1, jvmArgsAppend = { "-XX:+UseG1GC", "-Xss500m", "-Xms4g", "-Xmx4g", "-ea" })
+@Threads(1)
+public class TreeMapBenchmark {
+    private static final Ordering<Integer> SCALA_ORDERING = Ordering$.MODULE$.comparatorToOrdering(Integer::compareTo);
+
+    @Test
+    public void launchBenchmark() throws Exception {
+        Options opt = new OptionsBuilder()
+                .include(this.getClass().getSimpleName())
+                .shouldFailOnError(true)
+                .shouldDoGC(true)
+                .build();
+
+        Collection<RunResult> results = new Runner(opt).run();
+        BenchmarkPerformanceReporter reporter = BenchmarkPerformanceReporter.of(results);
+        reporter.printDetailedPerformanceReport();
+        reporter.printPerformanceRatiosReport();
+    }
+
+    // The JMH samples are the best documentation for how to use it
+    // http://hg.openjdk.java.net/code-tools/jmh/file/tip/jmh-samples/src/main/java/org/openjdk/jmh/samples/
+    @State(Scope.Thread)
+    public static class BenchmarkState {
+        @Param({"10", "1000", "100000"})
+        int numItems;
+
+        Random random = new Random();
+
+        int minKey;
+        int maxKey;
+
+        Array<Integer> keysSorted;
+        Array<Integer> keysRandom;
+        Array<Tuple2<Integer, String>> entriesRandom;
+
+        TreeMap<Integer, String> slangTreeMap = TreeMap.empty();
+        java.util.TreeMap<Integer, String> javaTreeMap = new java.util.TreeMap<>();
+        scala.collection.immutable.TreeMap<Integer, String> scalaTreeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+
+        @Setup(Level.Iteration)
+        public void setUp() {
+            minKey = 1;
+            maxKey = numItems * 2;
+
+            // We exclude keys with odd numbers to test methods requiring missing keys, such as floor or ceiling
+            keysSorted = Array.rangeBy(minKey, maxKey, 2);
+            keysRandom = randomize(keysSorted);
+            entriesRandom = keysRandom.map(i -> Tuple.of(i, Integer.toString(i)));
+
+            entriesRandom.forEach(e -> {
+                slangTreeMap = slangTreeMap.put(e._1, e._2);
```

Comment:
could you please put `mutable` and `persistent` in their names, so that we know when we're comparing apples to oranges? Also, it might make the benchmark quicker, if we don't rebuild the non-mutable ones